geopandas/tests/test_vectorized.py

```diff
@@ -0,0 +1,367 @@
+        result = getattr(T, attr)(other, *args)
+        assert isinstance(result, np.ndarray)
+        assert result.dtype == bool
+
+        expected = [getattr(tri, attr)(other, *args) for tri in triangles]
+
+        assert result.tolist() == expected
+
+
+@pytest.mark.parametrize('attr,args', [
+    ('contains', ()),
+    ('covers', ()),
+    ('crosses', ()),
+    ('disjoint', ()),
+    ('equals', ()),
+    ('intersects', ()),
+    ('overlaps', ()),
+    ('touches', ()),
+    ('within', ()),
+    ('equals_exact', (0.1,))
+])
+def test_vector_vector_predicates(attr, args):
+    A = [shapely.geometry.Polygon([(random.random(), random.random())
+                                   for i in range(3)])
+         for _ in range(100)]
+    B = [shapely.geometry.Polygon([(random.random(), random.random())
+                                   for i in range(3)])
+         for _ in range(100)]
+
+
+    vec_A = from_shapely(A)
+    vec_B = from_shapely(B)
+
+    result = getattr(vec_A, attr)(vec_B, *args)
+    assert isinstance(result, np.ndarray)
+    assert result.dtype == bool
+
+    expected = [getattr(a, attr)(b, *args) for a, b in zip(A, B)]
+
+    assert result.tolist() == expected
+
+
+@pytest.mark.parametrize('attr', [
+    'boundary',
+    'centroid',
+#    'representative_point',
+    'convex_hull',
+    'envelope',
+])
+def test_unary_geo(attr):
+
+    result = getattr(T, attr)()
+    expected = [getattr(t, attr) for t in triangles]
+
+    assert [a.equals(b) for a, b in zip(result, expected)]
+
+
+@pytest.mark.parametrize('attr', [
+    'difference',
+    'symmetric_difference',
+    'union',
+    'intersection',
+])
+def test_vector_binary_geo(attr):
+    quads = []
+    while len(quads) < 10:
+        geom = shapely.geometry.Polygon([(random.random(), random.random())
+                                        for i in range(4)])
+        if geom.is_valid:
+            quads.append(geom)
+
+    Q = from_shapely(quads)
+
+    result = getattr(T, attr)(Q)
+    expected = [getattr(t, attr)(q) for t, q in zip(triangles, quads)]
+
+    assert [a.equals(b) for a, b in zip(result, expected)]
+
+
+@pytest.mark.parametrize('attr', [
+    'difference',
+    'symmetric_difference',
+    'union',
+    'intersection',
+])
+def test_binary_geo(attr):
+    triangles = [shapely.geometry.Polygon([(random.random(), random.random())
+                                           for i in range(3)])
+                 for _ in range(10)]
+    quads = []
+    while len(quads) < 1:
+        geom = shapely.geometry.Polygon([(random.random(), random.random())
+                                        for i in range(4)])
+        if geom.is_valid:
+            quads.append(geom)
+
+    q = quads[0]
+
+    T = from_shapely(triangles)
+
+    result = getattr(T, attr)(q)
+    expected = [getattr(t, attr)(q) for t in triangles]
+
+    assert [a.equals(b) for a, b in zip(result, expected)]
+
+
+@pytest.mark.parametrize('attr', [
+    pytest.mark.xfail('is_closed'),
```

Comment:
I think it has to do with GEOS sometimes closing polygons.  I think the right thing to do here is to use LineStrings